sips/sip-xyz/sip-xyz-bitwise-ops.md

```diff
@@ -0,0 +1,195 @@
+# Preamble
+
+SIP Number: XYZ
+
+Title: Bitwise Operations in Clarity
+
+Authors: Cyle Witruk <https://github.com/cylewitruk>, Brice Dobry
+<https://github.com/obycode>
+
+Consideration: Technical, Governance
+
+Type: Consensus
+
+Status: Draft
+
+Created: 12 November 2022
+
+License: CC0-1.0
+
+Sign-off:
+
+Layer: Consensus (hard fork)
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP adds bitwise operations to the Clarity language which could simplify
+the implementation of smart contracts that require manipulation of bits. The
+changes include the addition of the following operations:
+
+- Bitwise Xor (`^`)
+- Bitwise And (`&`)
+- Bitwise Or (`|`)
+- Bitwise Not (`~`)
+- Binary Left Shift (`<<`)
+- Binary Right Shift (`>>`)
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0
+Universal license, available at
+https://creativecommons.org/publicdomain/zero/1.0/ This SIP's copyright is held
+by the Stacks Open Internet Foundation.
+
+# Introduction
+
+Bitwise operations are common in other programming languages. Common algorithms,
```

Comment:
My primary use case was simply being able to use `flags`, and easily setting/determining which flag was set.  Maybe also worth mentioning this here.

Reply:
Added in 1a19399b649aa81f81ada6345a2d659c0e3d0dcf